deployments/rialto/Front-end.Dockerfile

```
@@ -0,0 +1,22 @@
+FROM node:12 as build-deps
+
+ENV SUBSTRATE_PROVIDER wss://wss.rialto.brucke.link
+ENV ETHEREUM_PROVIDER http://rpc.rialto.brucke.link:8545
+ENV EXPECTED_ETHEREUM_NETWORK_ID 105
+
+# install tools and dependencies
+RUN set -eux; \
+	apt-get install -y git
+
+# clone UI repo
+RUN cd /usr/src/ && git clone https://github.com/paritytech/bridge-ui.git
+WORKDIR /usr/src/bridge-ui
+RUN yarn
+RUN yarn build:docker
```

```suggestion
ENV SUBSTRATE_PROVIDER wss://wss.rialto.brucke.link
ENV ETHEREUM_PROVIDER http://rpc.rialto.brucke.link:8545
ENV EXPECTED_ETHEREUM_NETWORK_ID 105
RUN yarn build:docker
```

This will allow the cache to be re-used for previous steps even if `ENV` changes.

---

Does the `ENV` variables always override `.env` files?

---

My goto document about those things is https://vsupalov.com/docker-arg-env-variable-guide/#setting-env-values
It says that ARG receive the `.env` values, and that you can use that to set the ENV. But setting the ENV directly should be fine and simply not take care about any `.env`. I guess this can be tested quickly once the UI PR are in.

---

I rather meant the UI build, not the `Docker`. To pass `ENV` to docker file build, you can put an `.env` file next to it, or you can just pass them via CLI indeed. But that's a docker thing. Then when we build UI we take the ENV variables that are set and we apply them to the code. We read another `.env` file (the one that is checked out from the repo) and my question was if running:
```
$ SUBSTRATE_PROVIDER=http://localhost yarn build:docker
```
causes the frontend `.env` file variable to be overwritten by the one provided in the CLI.

---

yes, the environment variable is overriding the `.env` e.g `SUBSTRATE_PROVIDER=blabla yarn build` resulted in a bundled js file containing "blabla".